apps/ex_wire/lib/ex_wire/sync/warp_processor.ex

```diff
@@ -0,0 +1,570 @@
+        end)
+
+      %{
+        state
+        | queued_block_chunks: [
+            {chunk_hash, block_chunk, pid} | state.queued_block_chunks
+          ]
+      }
+    end
+  end
+
+  # The task to process a block. When we process a block, we just verify the details
+  # about that block, store it's transactions, receipts and block data to the database,
+  # and return the processed data back to the sync process.
+  @spec new_block_chunk_task(pid(), EVM.hash(), BlockChunk.t(), Trie.t(), processor_mod(), pid()) ::
+          Task.t()
+  defp new_block_chunk_task(sup, chunk_hash, block_chunk, trie, processor_mod, pid) do
+    Task.Supervisor.async(sup, fn ->
+      start = Time.time_start()
+
+      :ok =
+        Logger.debug(fn ->
+          "[Warp] Starting to process #{Enum.count(block_chunk.block_data_list)} block(s)."
+        end)
+
+      {processed_blocks, block, next_trie} = processor_mod.process_block_chunk(block_chunk, trie)
+
+      trie_elapsed =
+        Time.elapsed(fn ->
+          TrieStorage.commit!(next_trie)
+        end)
+
+      :ok =
+        Logger.debug(fn ->
+          "[Warp] Processed #{Enum.count(processed_blocks)} block(s) in #{Time.elapsed(start)} (trie commit time: #{
+            trie_elapsed
+          })."
+        end)
+
+      :ok = GenServer.cast(pid, {:processed_block_chunk, chunk_hash, processed_blocks, block})
+
+      :ok
+    end)
+  end
+
+  # If we are below max parallelism, spawn a new state chunk processing task, or queue
+  # the state chunk if we've hit max parallelism.
+  @spec maybe_new_state_chunk_task(state(), pid(), EVM.hash(), StateChunk.t(), Trie.t(), pid()) ::
+          state()
+  defp maybe_new_state_chunk_task(
+         state = %{parallelism: parallelism, processor_mod: processor_mod},
+         sup,
+         chunk_hash,
+         state_chunk,
+         trie,
+         pid
+       ) do
+    if current_child_count(sup) <= parallelism do
+      _task = new_state_chunk_task(self(), sup, chunk_hash, state_chunk, trie, processor_mod, pid)
+
+      state
+    else
+      :ok =
+        Logger.debug(fn ->
+          "[Warp] Processor state queue size: #{Enum.count(state.queued_state_chunks) + 1}"
+        end)
+
+      %{
+        state
+        | queued_state_chunks: [
+            {chunk_hash, state_chunk, pid} | state.queued_state_chunks
+          ]
+      }
+    end
+  end
+
+  # The task to process a state chunk. When we process a state chunk, we decode the
+  # details about the included account entries. Then we pass the account state data
+  # back to this process to process into the state trie.
+  @spec new_state_chunk_task(
+          pid(),
+          pid(),
+          EVM.hash(),
+          StateChunk.t(),
+          Trie.t(),
+          processor_mod(),
+          pid()
+        ) :: Task.t()
+  defp new_state_chunk_task(processor_pid, sup, chunk_hash, state_chunk, trie, processor_mod, pid) do
+    Task.Supervisor.async(sup, fn ->
+      start = Time.time_start()
```

Comment:
Yup, I played around with this a few ways, but I wanted something to help us report times for sync since the process takes between a few minutes and a few hours and I really want to see `blocks/s`, etc, so I could figure out the rate. Over time, I plan to evolve this, but it seemed easy  enough to use our own little lib that had nicer reporting facilities.